core-primitives/node-api/api-client-extensions/src/chain.rs

```diff
@@ -19,55 +19,68 @@ use crate::ApiResult;
 use itp_types::{Header, SignedBlock};
 use sp_core::{storage::StorageKey, Pair, H256};
 use sp_finality_grandpa::{AuthorityList, VersionedAuthorityList, GRANDPA_AUTHORITIES_KEY};
+use sp_rpc::number::NumberOrHex;
 use sp_runtime::MultiSignature;
-use substrate_api_client::{Api, ExtrinsicParams, RpcClient};
+use substrate_api_client::{
+	Api, BalancesConfig, ExtrinsicParams, FrameSystemConfig, FromHexString, RpcClient,
+};
+
+use codec::Decode;
+use core::str::FromStr;
 
 pub type StorageProof = Vec<Vec<u8>>;
 
 /// ApiClient extension that simplifies chain data access.
-pub trait ChainApi {
+pub trait ChainApi<Runtime: FrameSystemConfig> {
```

Comment:
If you change the declaration of the trait like this:

```rust
pub trait ChainApi {
   type Hash;

   fn get_genesis_hash(&self) -> ApiResult<Self::Hash>
}
```

We can get rid of all the `<Runtime>`s, when we declare the trait bounds on the api, e.g:

```rust

Api: ChainApi<Runtime> -> Api: ChainApi

```
I would recommend this for the other api traits too.

Reply:
Note: I am only 99% sure that this works.

Reply:
I agree that's the cleaner solution, but do keep in mind that some of these traits are obsolete quite soon, because the api-client itself now (not yet on this version branch, but the next one, see https://github.com/scs/substrate-api-client/blob/7e5962bd2c9149f990b34aa25ad7bc7f6f92c868/src/api/rpc_api/chain.rs#L25-L31) implements traits. 
So, I think, it would be better to get rid of these rather sooner than later to reduce maintenance on the worker side. 
Summarizing: I don't think it's worth spending too much time on this trait.

Reply:
But that's only for the general traits. There are no traits incoming for the teeracle / teerex pallet.